desktop/tests/test_provider.py

```
@@ -37,6 +53,991 @@ def test_club_model_override(monkeypatch):
+    provider = DeepSeekProvider(api_key="secret", model=DEEPSEEK_MODEL)
+    emitted = []
+
+    async def consume():
+        async for chunk in provider.astream(
+            context_id="truncated",
+            messages=[{"role": "user", "content": "use a tool"}],
+            tools=[{"type": "function", "function": {"name": "now"}}],
+        ):
+            emitted.append(chunk)
+
+    with pytest.raises(RuntimeError, match="ended before.*DONE"):
```

Ruff (0.16.2)

[warning] Pattern passed to `match=` contains metacharacters but is neither escaped nor raw

(RUF043)